scripts/dependency-diagram.py

```
@@ -0,0 +1,200 @@
+
+def short(name: str) -> str:
+    """Strip the ``harmonia-`` prefix for readable node labels."""
+    return name.removeprefix(PREFIX)
+
+
+def get_workspace_edges(manifest_path: str | None = None) -> list[tuple[str, str]]:
+    """Return (from, to) pairs for intra-workspace normal dependencies."""
+    cmd = ["cargo", "metadata", "--format-version=1", "--no-deps"]
+    if manifest_path:
+        cmd += ["--manifest-path", manifest_path]
+    raw = subprocess.check_output(cmd, text=True)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)